src/FSharp.Data.GraphQL.Shared/Helpers/ObjAndStructConversions.fs

```diff
@@ -9,6 +9,18 @@ module internal ValueOption =
 
     let ofOption value = Option.toVOption value
 
+    let mergeWith (f: 'T -> 'T -> 'T) (o1 : 'T voption) (o2 : 'T voption) : 'T voption =
+        match (o1, o2) with
+        | ValueSome a, ValueSome b -> ValueSome (f a b)
+        | ValueSome a, _ -> ValueSome a
+        | _, ValueSome b -> ValueSome b
+        | _, _ -> ValueNone
+
+    let unwrap (defaultValue : 'U) (onSome : 'T -> 'U) (o : 'T voption) : 'U =
+        match o with
+        | ValueSome t -> onSome t
+        | ValueNone -> defaultValue
```

Comment:
I know this is not part of your changes, but this helper function is kind of suspect. Normally, a simple on-site pattern matching should be fine for this use case (instead of havint to use a helper function).